bindings/nodejs/src/wallet.rs

```diff
@@ -16,26 +16,52 @@ use crate::{
     build_js_error, client::ClientMethodHandler, destroyed_err, secret_manager::SecretManagerMethodHandler, NodejsError,
 };
 
-pub type WalletMethodHandler = Arc<RwLock<Option<Wallet>>>;
+pub struct WalletMethodHandlerInner(Option<Wallet>);
+
+impl Drop for WalletMethodHandlerInner {
+    fn drop(&mut self) {
+        log::debug!("drop WalletMethodHandlerInner");
+        // Request to stop the background syncing silently if this wallet hasn't been destroyed yet
+        if let Some(wallet) = self.0.take() {
+            wallet.request_stop_background_syncing();
```

Comment:
Don't you want to call this in `destroy_wallet()`?

Reply:
> Don't you want to call this in `destroy_wallet()`?

The point here is that you might not have control over how the wallet is destroyed, in our case this happens when reloading (ctrl-r) Firefly in development mode. The wallets instances get destroyed, but their destroyWallets methods never get called. This is kind of like a fallback so you don't have "zombie" threads locking the db.

Reply:
> Don't you want to call this in `destroy_wallet()`?

That would be great too though :) But it doesn't fix our issue